core/src/main/java/com/amplifyframework/analytics/Properties.java

```diff
@@ -15,34 +15,181 @@
 
 package com.amplifyframework.analytics;
 
-import com.amplifyframework.util.Immutable;
+import androidx.annotation.NonNull;
 
 import java.util.HashMap;
+import java.util.Iterator;
 import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Objects;
 
 /**
- * Wrapper for {@link Map} to restrict the types that may be passed in.
+ * Properties provides key/value pairs to represent attributes of an {@link AnalyticsEvent}. While
+ * different analytics services will have support for different datatypes, Properties provides a
+ * baseline of support for String, Boolean, Double, and Integer. If one of these aren't available in
+ * a given service, the plugin is expected to cast it to something supported. (e.g., convert Boolean
+ * true to the String "true")
+ *
+ * <pre>
+ *     Properties properties = Properties.builder()
+ *          .add("PostType", "UserImage")
+ *          .add("LikedUserID", 78219)
+ *          .add("FirstLike", true)
+ *          .build();
+ * </pre>
  */
-public class Properties {
+public final class Properties implements Iterable<Map.Entry<String, Property<?>>> {
```

Comment:
`AnalyticsProperties`? and each `Property` -> `AnalyticsProperty`?
- would we benefit having a less genericly named "Property" class and builder, in case "Property" is used else where in the future?
- by having this, developer experience would be:
```
Amplify.Analytics.registerGlobalProperties(
    AnalyticsProperties.builder()
        .add("Key1", "Value1")
        .add("Key2", "Value2")
        .build()
);
```